sdk/src/types/block/semantic/mod.rs

```diff
@@ -240,19 +270,63 @@ impl<'a> SemanticValidationContext<'a> {
                         Some(output.features()),
                     )
                 }
-                Output::Account(output) => (output.amount(), output.mana(), None, Some(output.features())),
+                Output::Account(output) => {
+                    if output.features().block_issuer().is_some() {
+                        let account_id = output.account_id_non_null(&OutputId::new(self.transaction_id, index as u16));
+
+                        if self.commitment_context_input.is_none() {
+                            return Ok(Some(TransactionFailureReason::BlockIssuerCommitmentInputMissing));
+                        }
+                        if !bic_context_inputs.contains(&account_id) {
+                            return Ok(Some(TransactionFailureReason::BlockIssuanceCreditInputMissing));
+                        }
+                        let entry = self.block_issuer_mana.entry(account_id).or_default();
+
+                        entry.1 = entry.1.checked_add(output.mana()).ok_or(Error::CreatedManaOverflow)?;
+
+                        if let Some(allotment) = self.transaction.allotments().get(&account_id) {
+                            entry.1 = entry
+                                .1
+                                .checked_add(allotment.mana())
+                                .ok_or(Error::CreatedManaOverflow)?;
+                        }
+                    }
+
+                    (output.amount(), output.mana(), None, Some(output.features()))
+                }
                 Output::Anchor(_) => return Err(Error::UnsupportedOutputKind(AnchorOutput::KIND)),
                 Output::Foundry(output) => (output.amount(), 0, output.native_token(), Some(output.features())),
                 Output::Nft(output) => (output.amount(), output.mana(), None, Some(output.features())),
                 Output::Delegation(output) => (output.amount(), 0, None, None),
             };
 
-            if let Some(sender) = features.and_then(|f| f.sender()) {
+            if let Some(sender) = features.and_then(Features::sender) {
                 if !self.unlocked_addresses.contains(sender.address()) {
                     return Ok(Some(TransactionFailureReason::SenderFeatureNotUnlocked));
                 }
             }
 
+            if let Some(unlock_conditions) = created_output.unlock_conditions() {
+                if let (Some(address), Some(timelock)) = (unlock_conditions.address(), unlock_conditions.timelock()) {
+                    if let Address::Account(account_address) = address.address() {
+                        if let Some(entry) = self.block_issuer_mana.get_mut(account_address.account_id()) {
+                            let past_bounded_slot_index = self
+                                .protocol_parameters
+                                .past_bounded_slot(self.commitment_context_input.unwrap());
```

Comment:
do we need to document that `unwrap`?

Reply:
Or needs to return `TransactionFailureReason::TimelockCommitmentInputMissing`

Reply:
Turns out the unwrap wasn't really justified, but I have used `BlockIssuerCommitmentInputMissing` instead